proof/courier.go

```diff
@@ -854,6 +926,195 @@ func (h *HashMailCourier) SetSubscribers(
 // proof.Courier interface.
 var _ Courier = (*HashMailCourier)(nil)
 
+// UniverseRpcCourier is a universe RPC proof courier service handle. It
+// implements the Courier interface.
+type UniverseRpcCourier struct {
+	// recipient describes the recipient of the proof.
+	recipient Recipient
+
+	// client is the RPC client that the courier will use to interact with
+	// the universe RPC server.
+	client unirpc.UniverseClient
+
+	// deliveryLog is the log that the courier will use to record the
+	// attempted delivery of proofs to the receiver.
+	deliveryLog DeliveryLog
+
+	// subscribers is a map of components that want to be notified on new
+	// events, keyed by their subscription ID.
+	subscribers map[uint64]*fn.EventReceiver[fn.Event]
+
+	// subscriberMtx guards the subscribers map and access to the
+	// subscriptionID.
+	subscriberMtx sync.Mutex
+}
+
+// DeliverProof attempts to delivery a proof file to the receiver.
+func (c *UniverseRpcCourier) DeliverProof(ctx context.Context,
+	annotatedProof *AnnotatedProof) error {
+
+	// Decode annotated proof into proof file.
+	proofFile := &File{}
+	err := proofFile.Decode(bytes.NewReader(annotatedProof.Blob))
+	if err != nil {
+		return err
+	}
+
+	// Iterate over each proof in the proof file and submit to the courier
+	// service.
+	for i := range proofFile.proofs {
```

Comment:
I think we should use the proof file's helper methods here, like `proofFile.NumProofs()` and `proofFile.ProofAt(idx)`. Then we don't need to do some of the decoding ourselves.